docs/proposals/mapper- provide-a- common- MQTT- mapper.md

```diff
@@ -0,0 +1,277 @@
+
+##### Base Environment Setup
+
+1. Generate Mapper files through the Mapper-Framework framework
+   
+   1)Download kubeedge-master.zip from the official website, take the mapper-framewokr under the staging folder in it, and transfer it to the VM master node.
+   
+   2)Execute a command: 
+   
+   ```go
+   make generate
+   ```
+   
+   3)Enter the project name of the Mapper)
+   
+   <img title="" src="../images/proposals/mapper-provide-a-common-MQTT-mapper-pic.jpg" alt="" data-align="center">
+
+2. Determine the Device model and Device instance files for device
+   
+   1)Device model: A device model describes the device properties exposed by the device . A device model is a Physical model which constrains the properties and parameters of physical devices.
+   
+   ```
+   apiVersion: devices.kubeedge.io/v1beta1
+   kind: DeviceModel
+   metadata:
+     name: beta1-model
+   spec:
+     properties:
+       - name: temp
+         description: beta1-model
+         type: INT
+         accessMode: ReadWrite
+         maximum: "100"
+         minimum: "1"
+         unit: "Celsius"
+     protocol: modbus
+   ```
+   
+   2)Device instance: A device instance represents an actual device object. The device spec is static, including device properties list, it describes the details of each property, including its name, type, access method. The device status contains dynamically changing data like the desired state of a device property and the state reported by the device.
+   
+   ```
+   apiVersion: devices.kubeedge.io/v1beta1
+   kind: Device
+   metadata:
+     name: beta1-device
+   spec:
+     deviceModelRef:
+       name: beta1-model
+     nodeName: worker-node1
+     properties:
+       - name: temp
+         collectCycle: 10000000000  # The frequency of reporting data to the cloud, once every 10 seconds
+         reportCycle: 10000000000   # The frequency of data push to user applications or databases, once every 10 seconds
+         reportToCloud: true
+         desired:
+           value: "30"
+         pushMethod:
+           mqtt:
+             address: tcp://101.133.150.110:1883
+             topic: temp
+             qos: 0
+             retained: false
+           dbMethod:
+             influxdb2:
+               influxdb2ClientConfig:
+                 url: http://127.0.0.1:8086
+                 org: test-org
+                 bucket: test-bucket
+               influxdb2DataConfig:
+                 measurement: stat
+                 tag:
+                   unit: temperature
+                 fieldKey: beta1test
+         visitors:
+           protocolName: modbus
+           configData:
+             register: "HoldingRegister"
+             offset: 2
+             limit: 1
+             scale: 1
+             isSwap: true
+             isRegisterSwap: true
+     protocol:
+       protocolName: modbus
+       configData:
+         ip: 172.17.0.3
+         port: 1502
```

Comment:
ditto

Reply:
Ok, I'll all change the model and instance files based on your comments.